pkg/dockerregistry/server/configuration/configuration.go

```diff
@@ -3,18 +3,71 @@ package configuration
 import (
 	"bytes"
 	"errors"
+	"fmt"
 	"io"
 	"io/ioutil"
 	"os"
 	"reflect"
 	"strings"
 	"time"
 
+	log "github.com/Sirupsen/logrus"
 	"gopkg.in/yaml.v2"
 
+	//"github.com/docker/distribution/registry/auth"
 	"github.com/docker/distribution/configuration"
 )
 
+// Environment variables.
+const (
+	// DockerRegistryURLEnvVar is a mandatory environment variable name specifying url of internal docker
+	// registry. All references to pushed images will be prefixed with its value.
+	// DEPRECATED: Use the OPENSHIFT_DEFAULT_REGISTRY instead.
+	DockerRegistryURLEnvVar = "DOCKER_REGISTRY_URL"
+
+	// DockerRegistryURLEnvVarOption is an optional environment that overrides the
+	// DOCKER_REGISTRY_URL.
+	DockerRegistryURLEnvVarOption = "REGISTRY_MIDDLEWARE_REPOSITORY_OPENSHIFT_DOCKERREGISTRYURL"
+
+	// OpenShiftDefaultRegistry overrides the DockerRegistryURLEnvVar as in OpenShift the
+	// default registry URL is controller by this environment variable.
```

Comment:
s/controller/controlled/

This also overrides the value in the config file, right?  (all these env vars override values in the config file)

Reply:
@bparees yes.

Reply:
ok so let's make sure the godc states that for all these values.